yb-voyager/src/tgtdb/suites/yugabytedbSuite_test.go

```diff
@@ -148,3 +148,1523 @@ func TestBytesConversionWithFormatting(t *testing.T) {
+		}
+		inputStr := string(input)
+
+		// Convert with formatIfRequired=true (SQL format with quotes)
+		result, err := YBValueConverterSuite["STRING"](inputStr, true, nil)
+		assert.NoError(t, err)
+
+		// Verify result
+		expectedSize := size + 2 // +2 for quotes
+		assert.Equal(t, expectedSize, len(result), "10KB + 2 quotes = 10242 bytes")
+		assert.Equal(t, '\'', rune(result[0]), "Should start with quote")
+		assert.Equal(t, '\'', rune(result[len(result)-1]), "Should end with quote")
+		assert.Equal(t, inputStr, result[1:len(result)-1], "Content should match")
+
+		t.Logf("✓ 10KB string: %d bytes → %d bytes", size, len(result))
+	})
+
+	// Test 2: 10MB string (large document size)
+	t.Run("10MB string", func(t *testing.T) {
+		size := 10 * 1024 * 1024 // 10MB = 10485760 bytes
+		input := make([]byte, size)
+		for i := range input {
+			input[i] = byte('a' + (i % 26))
+		}
+		inputStr := string(input)
+
+		// Convert with formatIfRequired=true
+		result, err := YBValueConverterSuite["STRING"](inputStr, true, nil)
+		assert.NoError(t, err)
+
+		// Verify length (full content check would be slow)
+		expectedSize := size + 2 // +2 for quotes
+		assert.Equal(t, expectedSize, len(result), "10MB + 2 quotes = 10485762 bytes")
+
+		// Verify first and last 1KB (data integrity spot check)
+		assert.Equal(t, inputStr[:1024], result[1:1025], "First 1KB should match")
+		assert.Equal(t, inputStr[len(inputStr)-1024:], result[len(result)-1025:len(result)-1], "Last 1KB should match")
+
+		t.Logf("✓ 10MB string: %d bytes → %d bytes", size, len(result))
+	})
+
+	// Test 3: 200MB string (extreme stress test - matches BYTES test)
+	t.Run("200MB string", func(t *testing.T) {
+		size := 200 * 1024 * 1024 // 200MB = 209715200 bytes
+		input := make([]byte, size)
+		for i := range input {
+			input[i] = byte('a' + (i % 26))
+		}
+		inputStr := string(input)
+
+		t.Logf("Converting 200MB string...")
+		start := time.Now()
+		result, err := YBValueConverterSuite["STRING"](inputStr, true, nil)
+		duration := time.Since(start)
+
+		assert.NoError(t, err)
+
+		// Verify length
+		expectedSize := size + 2 // +2 for quotes
+		assert.Equal(t, expectedSize, len(result), "200MB + 2 quotes = 209715202 bytes")
+
+		// Verify first and last 1KB (data integrity spot check)
+		assert.Equal(t, inputStr[:1024], result[1:1025], "First 1KB should match")
+		assert.Equal(t, inputStr[len(inputStr)-1024:], result[len(result)-1025:len(result)-1], "Last 1KB should match")
+
+		t.Logf("✓ 200MB string: %d bytes → %d bytes in %v", size, len(result), duration)
+		t.Logf("✓ Throughput: %.2f MB/s", float64(size)/(1024*1024)/duration.Seconds())
+	})
+}
+
+// Test 2.1: TestJsonConversionBasic
+// Tests JSON/JSONB type with basic valid JSON (no single quotes inside JSON strings)
+// NOTE: The converter uses quoteValueIfRequiredWithEscaping which doubles single quotes,
+// but this creates INVALID JSON. Therefore, we only test JSON that doesn't contain
```

Comment:
why does this matter? we just need to assert it properly, why does it matter if it's an invalid json or not